crate does not re-export `bstr`, therefore requires users to add this dependency

hey, on my machine this example

```rust
use unfmt::unformat;

fn main() {
    unformat!("hello", world);
}
```
in `Cargo.toml`:
```toml
[dependencies]
unfmt = { git = "https://github.com/mathematic-inc/unfmt.git" }
```

...results in
```
error[E0432]: unresolved import `bstr`
```

I believe this is because the macro expansion includes `use ::bstr::{ByteSlice, BStr};`.
this means a user would need to add `bstr` as a dependency - see [this Stack Overflow question](https://stackoverflow.com/questions/54593472/how-do-i-write-a-procedural-macro-that-expands-to-a-macro-invocation-without-req).

afaik, there are ways to deal with this, such as re-exporting.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,16 +10,19 @@ categories = ["no-std", "value-formatting"]
 keywords = ["unformat", "regex"]
 
 [lib]
-proc-macro = true
+path = "./unfmt.rs"
 
-[lints.clippy]
+[workspace]
+members = ["unfmt_macros"]
+
+[workspace.lints.clippy]
 all = "deny"
 pedantic = "deny"
 restriction = "deny"
 nursery = "deny"
 # REASON: We disable them when they are not idiomatic.
 blanket_clippy_restriction_lints = { level = "allow", priority = 1 }
-# REAsON: Not idiomatic.
+# REASON: Not idiomatic.
 implicit_return = { level = "allow", priority = 1 }
 # REASON: False-positives with macros.
 pub_use = { level = "allow", priority = 1 }
@@ -58,6 +61,4 @@ needless_borrowed_reference = { level = "allow", priority = 1 }
 
 [dependencies]
 bstr = "1"
-syn = "2"
-quote = "1"
-proc-macro2 = "1"
+unfmt_macros = { path = "unfmt_macros" }

--- a/tests/e2e.rs
+++ b/tests/e2e.rs
@@ -1,0 +1,54 @@
+use std::{
+    env::{self, temp_dir},
+    fs::{create_dir_all, write},
+};
+
+#[test]
+fn test_e2e() {
+    let e2e_dir = temp_dir().join("unfmt_e2e");
+    create_dir_all(e2e_dir.join("src")).expect("failed to create temp dir");
+
+    write(
+        e2e_dir.join("src/main.rs"),
+        r#"use unfmt::unformat;fn main() {unformat!("hello {}", "hello world");}"#,
+    )
+    .expect("failed to write file");
+
+    let mut cargo_dir = env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR not set");
+    if cfg!(windows) {
+        cargo_dir = cargo_dir.replace("\\", "/");
+    }
+    write(
+        e2e_dir.join("Cargo.toml"),
+        format! { r#"
+[package]
+name = "unfmt_e2e"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies.unfmt]
+path = "{cargo_dir}"
+"#},
+    )
+    .expect("failed to write file");
+
+    let output = std::process::Command::new("cargo")
+        .arg("run")
+        .current_dir(&e2e_dir)
+        .output()
+        .expect("failed to run cargo");
+
+    if !output.status.success() {
+        println!("tempdir: {}", e2e_dir.display());
+        println!(
+            "stderr: {}",
+            std::str::from_utf8(&output.stderr).expect("failed to convert stdout to string")
+        );
+        println!(
+            "stdout: {}",
+            std::str::from_utf8(&output.stdout).expect("failed to convert stdout to string")
+        );
+    }
+
+    assert!(output.status.success());
+}

--- a/unfmt.rs
+++ b/unfmt.rs
@@ -1,0 +1,2 @@
+pub use bstr;
+pub use unfmt_macros::*;

--- a/unfmt_macros/Cargo.toml
+++ b/unfmt_macros/Cargo.toml
@@ -1,0 +1,23 @@
+[package]
+name = "unfmt_macros"
+version = "0.1.0"
+edition = "2021"
+description = "A compile-time pattern matching library that reverses the interpolation process of `format!`."
+license = "MIT OR Apache-2.0"
+authors = ["Mathematic Inc"]
+repository = "https://github.com/mathematic-inc/unfmt"
+categories = ["no-std", "value-formatting"]
+keywords = ["unformat", "regex"]
+
+[lib]
+proc-macro = true
+path = "./unfmt_macros.rs"
+
+[lints]
+workspace = true
+
+[dependencies]
+bstr = "1"
+syn = "2"
+quote = "1"
+proc-macro2 = "1"

--- a/unfmt_macros/unfmt_macros.rs
+++ b/unfmt_macros/unfmt_macros.rs
@@ -254,8 +254,8 @@ pub fn unformat(input: TokenStream) -> TokenStream {
 
     TokenStream::from(quote! {
         'unformat: {
-            use ::std::str::FromStr;
-            use ::bstr::{ByteSlice, BStr};
+            use ::core::str::FromStr;
+            use ::unfmt::bstr::{ByteSlice, BStr};
             let Some((__unfmt_left, mut __unfmt_byte_text)) = BStr::new(#text).split_once_str(#initial_part) else {
                 break 'unformat None;
             };

